src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -85,10 +86,25 @@
     @Override
     public boolean start() throws Exception {
         final PlaceholderTask task = new PlaceholderTask(getContext(), step.getLabel(), run);
-        if (Queue.getInstance().schedule2(task, 0).getCreateItem() == null) {
+        Queue.Item taskInQueue = Queue.getInstance().schedule2(task, 0).getCreateItem();
+        if (taskInQueue == null) {
             // There can be no duplicates. But could be refused if a QueueDecisionHandler rejects it for some odd reason.
             throw new IllegalStateException("failed to schedule task");
         }
+        String initialWhyBlocked = taskInQueue.getWhy();
+        if (initialWhyBlocked == null) {
+            // Reuse the general
+            initialWhyBlocked = hudson.model.Messages.Queue_Unknown();
+        }
+
+        ExecutorTaskInfoAction taskAction = flowNode.getAction(ExecutorTaskInfoAction.class);
+        if (taskAction == null) {
+            // Populate the initial version of the ExecutorTaskInfoAction.
+            flowNode.addAction(new ExecutorTaskInfoAction(initialWhyBlocked, flowNode));
+        } else {
```

Comment:
`onResume()` potentially calls `start()` - I decided to play it safe for that scenario.